pkg/controller/kubedirectorcluster/members.go

```diff
@@ -280,10 +317,65 @@ func handleReadyMembers(
 				)
 				return
 			}
+
+			memberVersion := *m.StateDetail.LastConnectionVersion
+
+			shared.LogInfo(
+				reqLogger,
+				cr,
+				shared.EventReasonCluster,
+				fmt.Sprintf("memberVersion for pod %s is %d", m.Pod, memberVersion),
+			)
+
+			if memberVersion < connectionsVersion {
+				shared.LogInfo(
+					reqLogger,
+					cr,
+					shared.EventReasonCluster,
+					fmt.Sprintf("--reconnect will be called here for pod : %s", m.Pod),
+				)
+
+				containerID := m.StateDetail.LastConfiguredContainer
+				cmd := fmt.Sprintf(appPrepConfigReconnectCmd, containerID)
+
+				shared.LogInfo(
+					reqLogger,
+					cr,
+					shared.EventReasonCluster,
+					fmt.Sprintf("Executing command %s", cmd),
+				)
+				cmdErr := executor.RunScript(
+					reqLogger,
+					cr,
+					cr.Namespace,
+					m.Pod,
+					m.StateDetail.LastConfiguredContainer,
+					executor.AppContainerName,
+					"app config",
+					strings.NewReader(cmd),
+				)
+				if cmdErr != nil {
+					shared.LogErrorf(
+						reqLogger,
+						cmdErr,
+						cr,
+						shared.EventReasonMember,
+						"failed to run startcsript with --reconnect in member{%s} in role{%s}",
+						m.Pod,
+						role.roleStatus.Name,
+					)
+					return
+				}
```

Comment:
How are we going to handle the failure case here? If the run script fails, the member version would still not be equal to the current connection version, we would retry?

Reply:
The behaviour will be consistent with what happens when an error is encountered while injecting configmeta.json in the pod. An error would be logged and the flow would return without retrying again.